eclair-core/src/main/scala/fr/acinq/eclair/crypto/keymanager/CommitmentKeys.scala

```diff
@@ -0,0 +1,136 @@
+      theirPaymentPublicKey = params.commitmentFormat match {
+        case DefaultCommitmentFormat if params.channelFeatures.hasFeature(Features.StaticRemoteKey) => params.remoteParams.paymentBasepoint
+        case DefaultCommitmentFormat => ChannelKeys.remotePerCommitmentPublicKey(params.remoteParams.paymentBasepoint, localPerCommitmentPoint)
+        case _: AnchorOutputsCommitmentFormat => params.remoteParams.paymentBasepoint
+      },
+      ourPaymentBasePoint = params.localParams.walletStaticPaymentBasepoint.getOrElse(channelKeys.paymentBasePoint),
+      ourHtlcKey = channelKeys.htlcKey(localPerCommitmentPoint),
+      theirHtlcPublicKey = ChannelKeys.remotePerCommitmentPublicKey(params.remoteParams.htlcBasepoint, localPerCommitmentPoint),
+      revocationPublicKey = ChannelKeys.revocationPublicKey(params.remoteParams.revocationBasepoint, localPerCommitmentPoint)
+    )
+  }
+}
+
+/**
+ * Keys used for the remote commitment.
+ * WARNING: these private keys must never be stored on disk, in a database, or logged.
+ *
+ * There is a subtlety for [[ourPaymentKey]]: when using option_static_remotekey, our output will directly send funds
+ * to a p2wpkh address created by our bitcoin node. We thus don't need the private key, as the output can immediately
+ * be spent by our bitcoin node (no need for 2nd-stage transactions to send it back to our wallet).
+ */
+case class RemoteCommitmentKeys(ourPaymentKey: Either[PublicKey, PrivateKey],
+                                theirDelayedPaymentPublicKey: PublicKey,
+                                ourPaymentBasePoint: PublicKey,
+                                ourHtlcKey: PrivateKey,
+                                theirHtlcPublicKey: PublicKey,
+                                revocationPublicKey: PublicKey) {
+  val ourPaymentPublicKey: PublicKey = ourPaymentKey match {
+    case Left(publicKey) => publicKey
+    case Right(privateKey) => privateKey.publicKey
+  }
+  // Since this is the remote commitment, local is them and remote is us.
+  val publicKeys: CommitmentPublicKeys = CommitmentPublicKeys(
+    localDelayedPaymentPublicKey = theirDelayedPaymentPublicKey,
+    remotePaymentPublicKey = ourPaymentKey match {
+      case Left(publicKey) => publicKey
+      case Right(privateKey) => privateKey.publicKey
+    },
+    localHtlcPublicKey = theirHtlcPublicKey,
+    remoteHtlcPublicKey = ourHtlcKey.publicKey,
+    revocationPublicKey = revocationPublicKey
+  )
+}
+
+object RemoteCommitmentKeys {
+  def apply(params: ChannelParams, channelKeys: ChannelKeys, remotePerCommitmentPoint: PublicKey): RemoteCommitmentKeys = {
+    RemoteCommitmentKeys(
+      ourPaymentKey = params.localParams.walletStaticPaymentBasepoint match {
+        case Some(walletPublicKey) => Left(walletPublicKey)
+        case None => params.commitmentFormat match {
+          case DefaultCommitmentFormat if params.channelFeatures.hasFeature(Features.StaticRemoteKey) => Right(channelKeys.paymentBaseSecret)
```

Comment:
This should not be possible (`walletStaticPaymentBasepoint == None` implies that `Features.StaticRemoteKey` is not set) and we did not have this branch before, why is it needed  ?